sdk/translation/ai-translation-document-rest/samples-dev/batchDocumentTranslation.ts

```diff
@@ -1,46 +1,118 @@
-  const client = createClient(endpoint, credentials);
+  const credential = new DefaultAzureCredential();
+  const client = createClient(endpoint, credential);
+  const blobServiceClient = new BlobServiceClient(blobEndpoint, credential);
+  const sourceContainerClient = blobServiceClient.getContainerClient("docs");
+  await sourceContainerClient.createIfNotExists();
+  const sourceBlobName = "file.pdf";
+  const sourceBlobClient = sourceContainerClient.getBlobClient(sourceBlobName);
+  const blockBlobClient = sourceBlobClient.getBlockBlobClient();
 
-  const sourceUrl = await createSourceContainer(ONE_TEST_DOCUMENTS);
-  const sourceInput = createSourceInput(sourceUrl);
-  const targetUrl = await createTargetContainer();
-  const targetInput = createTargetInput(targetUrl, "fr");
-  const batchRequest = createBatchRequest(sourceInput, [targetInput]);
+  await new Promise<void>((resolve, reject) => {
+    https
+      .get(fileUrl, (response) => {
+        if (response.statusCode !== 200) {
+          reject(new Error(`Failed to fetch file. Status code: ${response.statusCode}`));
+          return;
+        }
+        blockBlobClient
+          .uploadStream(response)
+          .then(() => resolve())
+          .catch(reject);
+      })
+      .on("error", reject);
+  });
+
+  const targetContainerClient = blobServiceClient.getContainerClient("translations");
+  await targetContainerClient.createIfNotExists();
 
   // Start translation
-  const batchRequests = { inputs: [batchRequest] };
-  const poller = await client.path("/document/batches").post({
-    body: batchRequests,
+  const response = await client.path("/document/batches").post({
+    body: {
+      inputs: [
+        {
+          source: {
+            sourceUrl: sourceContainerClient.url,
+          },
+          targets: [
+            {
+              targetUrl: targetContainerClient.url,
+              language: "fr",
+            },
+          ],
+        },
+      ],
+    },
+  });
+  if (isUnexpected(response)) {
+    throw createRestError(response);
+  }
+
+  const poller = await getLongRunningPoller(client, response, {
+    intervalInMs: 5000,
+    updateState: (_, pollRes) => {
+      process.stdout.write("Translation status: ");
+      console.log(JSON.stringify((pollRes.flatResponse as any).body, null, 2));
+    },
   });
-  if (isUnexpected(poller)) {
-    throw poller.body;
+  const result = await poller.pollUntilDone();
+  if (isUnexpected(result)) {
+    throw createRestError(result);
+  }
+  console.log("Translation completed successfully.");
+
+  const __filename = fileURLToPath(import.meta.url);
+  const __dirname = path.dirname(__filename);
+
+  for await (const blob of targetContainerClient.listBlobsFlat()) {
+    const translatedBlobClient = targetContainerClient.getBlobClient(blob.name);
+    const downloadResponse = await translatedBlobClient.download();
+    const localFilePath = path.join(__dirname, "downloadedTranslatedFile.pdf");
```

Comment:
should we include blob names in the downloaded local file names? Otherwise the same file is overwritten. Also, blob client has a `downloadToFile()` for convenience.

Reply:
Great catch, addressed in https://github.com/Azure/azure-sdk-for-js/pull/33640/commits/18ae14fd1dd8b4ebdd54c26850cf62a9b3db536a